UKHomeOffice/sgmr-service
# Cg 121 vessel page

NOTES : have added an error for missing voyageId scenario, need to discuss with Andrew/TomC to agree what this should be and update it

TO TEST

- Go to /vessels
- Go to save a vessel
- Fill out the fields (most are optional) and click save
- Make sure you have at least 2 saved vessels before continuing

**I can see my saved vessels on the voyage form**
- Go to /reports
- Go to start now
- Complete the departure page (only date is required)
- Click save and continue
- Complete the arrival page (nothing is required)
- Click save and continue
> you should see a list of your saved vessels as well as a vessel form

**I can select only one vessel at a time**
- Tick one checkbox
- Tick the other checkbox
> you should see them behave as though they’re radio buttons and you can only select one at this time

**I can add a saved vessel to my voyage**
- Tick one checkbox
- Click Add to report
>You should see your vessel details in the form below
- Click save and continue
>You should be taken to a blank page 4
> you should see your vessel added to your voyage data in network tab

**I can add a new vessel**
- On vessel page
- Do NOT tick a checkbox
- Fill in some or all of the vessel form
- Click save and continue
> you should see your vessel added to your voyage data in network tab
> you should not see this vessel listed on the /vessel page

**I can change details of a saved vessel just for this voyage**
- On vessel page
- Tick one checkbox and click add to report
- Change one or more fields on the form 
- Click save and continue
> you should see your vessel added to your voyage data in network tab
> you should not see the changes to the vessel listed on your /vessel page

**If something breaks (missing voyageId) I know what happened**
- Go directly to http://localhost:8080/save-voyage/page-3 
- Try to save and continue
>you should see a page error advising of the problem

**If I click add to report without a checkbox checked, nothing should happen**
- Click add report without checking a checkbox
>you should see nothing happens

----

**I can edit the vessel connected to the voyage**
- go to /reports
- click on a vessel name/departure date
> you should see page 6 with deprature, arrival, and vessel details populated
- click on change next to vessel details
> you should see vessel page with the form populated with your voyage vessel data

- **I can replace vessel with a saved vessel**
- - check a checkbox
-  - click add to voyage
> you should see the checked vessel details in the form
-  - click save and continue
> you should see a blank page 4
-  - return to /reports
-  - click on vessel name
> you should see the new vessel details 

- **I can update vessel details for this voyage**
- - add/edit/delete data from the form fields
- - click save and continue
> you should see a blank page 4
-  - return to /reports
-  - click on vessel name
> you should see the new vessel details 


NOTES: while there is validation on the vessel form (as it uses the same as the new vessel form) it does not yet handle you correcting errors by ticking a vessel and clicking add to report. This will be corrected in a different PR
